exchange/auction.go

```diff
@@ -88,61 +111,56 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 	for _, imp := range bidRequest.Imp {
 		expByImp[imp.ID] = imp.Exp
 	}
-	for _, topBidsPerImp := range a.winningBidsByBidder {
-		for _, topBidPerBidder := range topBidsPerImp {
-			impID := topBidPerBidder.bid.ImpID
-			isOverallWinner := a.winningBids[impID] == topBidPerBidder
-			if !includeBidderKeys && !isOverallWinner {
-				continue
+
+	/* bidders vs winners			*/
+	if targData.includeBidderKeys {
+		// cache banners and/or videos found in impsToBiddersTopBids map[string]map[openrtb_ext.BidderName]*pbsOrtbBid
+		for impID := range a.impsToBiddersTopBids {
+			for _, bid_to_cache := range a.impsToBiddersTopBids[impID] {
+				bids_to_cache_arr = append(bids_to_cache_arr, bid_to_cache)
 			}
-			var customCacheKey string
-			var catDur string
-			useCustomCacheKey := false
-			if competitiveExclusion && isOverallWinner {
-				// set custom cache key for winning bid when competitive exclusion applies
-				catDur = bidCategory[topBidPerBidder.bid.ID]
-				if len(catDur) > 0 {
-					customCacheKey = fmt.Sprintf("%s_%s", catDur, hbCacheID)
-					useCustomCacheKey = true
-				}
+		}
+	} else if targData.includeWinners {
```

Comment:
What if `targData.includeWinners` and `targData.incudeBidderKeys` are both true?

Reply:
When `targData.incudeBidderKeys` is true, we will also include winners in the `bids_to_cahce_arr` therefore, even if we don't go into the true branch of the `else if` statement in line 123, those bids will be copied into `bids_to_cahce_arr` because `impsToBiddersTopBids`, which was traversed in line 118, includes both regular bids and top bids.

Reply:
The above line can simply be
} else {
because we know targData.includeWinners is true if we get here.

I think Hans' comment above indicates that the comments in the code could be improved.  I believe the point is that when includeBidderKeys is set, we need to cache all bids but, otherwise, we need only cache the winners.  I suggest, in the interests of readability, changing the comment at line 117 (and maybe 124) to reflect that and eliminating the comment at line 115, or eliminating 117 and 124 entirely and replacing with a very descriptive 115.

Reply:
Done